### PR TITLE
chore: Replacing use of hard-coded constants with curve and duration tokens from theme

### PR DESCRIPTION
**Pull request by khmakoto**
## PR Description

This PR replaces the use of hard-coded constants for animation curves and durations with their respective tokens from `@fluentui/react-theme`.

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25522/perf-test-react-components/Avatar-mount_base.html">1606</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25522/perf-test-react-components/Avatar-mount.html">1657</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25522/perf-test-react-components/Button-mount_base.html">1167</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25522/perf-test-react-components/Button-mount.html">1179</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25522/perf-test-react-components/FluentProvider-mount_base.html">1971</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25522/perf-test-react-components/FluentProvider-mount.html">1975</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25522/perf-test-react-components/FluentProviderWithTheme-mount_base.html">778</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25522/perf-test-react-components/FluentProviderWithTheme-mount.html">797</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25522/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">739</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25522/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">714</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25522/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">781</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25522/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">772</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25522/perf-test-react-components/MakeStyles-mount_base.html">2344</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25522/perf-test-react-components/MakeStyles-mount.html">2353</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25522/perf-test-react-components/SpinButton-mount_base.html">3095</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25522/perf-test-react-components/SpinButton-mount.html">3123</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                                                             |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                   Change |
| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                                                                       |  `84.683 kB`<br />`21.252 kB` |   `84.755 kB`<br />`21.27 kB` |   `72 B` <br />`18 B`  |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                                                                    |  `48.874 kB`<br />`13.864 kB` |  `48.956 kB`<br />`13.883 kB` |   `82 B` <br />`19 B`  |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr>                                                                  |  `63.452 kB`<br />`17.959 kB` |  `63.534 kB`<br />`17.979 kB` |   `82 B` <br />`20 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                                                                    |   `36.844 kB`<br />`9.927 kB` |   `36.834 kB`<br />`9.913 kB` | `-10 B` <br />`-14 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                                                                    |  `43.869 kB`<br />`11.144 kB` |   `43.859 kB`<br />`11.13 kB` | `-10 B` <br />`-14 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                                                            |  `39.538 kB`<br />`10.826 kB` |  `39.528 kB`<br />`10.811 kB` | `-10 B` <br />`-15 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                                                                          |  `46.984 kB`<br />`12.211 kB` |  `46.974 kB`<br />`12.196 kB` | `-10 B` <br />`-15 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                                                                        |  `52.529 kB`<br />`11.411 kB` |  `52.519 kB`<br />`11.396 kB` | `-10 B` <br />`-15 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `62.935 kB`<br />`17.686 kB` |  `62.925 kB`<br />`17.674 kB` | `-10 B` <br />`-12 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `189.901 kB`<br />`52.991 kB` | `189.891 kB`<br />`52.966 kB` | `-10 B` <br />`-25 B`  |
| <samp>react-persona</samp> <br /> <abbr title='bundle-size/Persona.fixture.js'>Persona</abbr>                                                                                 |   `53.992 kB`<br />`15.25 kB` |  `54.074 kB`<br />`15.269 kB` |   `82 B` <br />`19 B`  |
| <samp>react-spinner</samp> <br /> <abbr title='bundle-size/Spinner.fixture.js'>Spinner</abbr>                                                                                 |   `20.023 kB`<br />`6.461 kB` |   `20.003 kB`<br />`6.449 kB` | `-20 B` <br />`-12 B`  |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr>                                                                                    |  `33.388 kB`<br />`10.578 kB` |  `33.366 kB`<br />`10.566 kB` | `-22 B` <br />`-12 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                  |         Size (minified/GZIP) |
| -------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr>                                               |  `14.996 kB`<br />`6.013 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>                                                     | `71.863 kB`<br />`20.939 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>                                                               | `67.641 kB`<br />`19.897 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>                                                   |   `8.607 kB`<br />`3.622 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>                                                   |   `9.782 kB`<br />`3.998 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr>                                                 |   `8.708 kB`<br />`3.678 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `33.446 kB`<br />`11.033 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                              |   `5.857 kB`<br />`1.978 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/176464a38a9b6210505d6950e8bd81df959427e0'>176464a38a9b6210505d6950e8bd81df959427e0</a></sub>
<div id="bundleSizeComment9423"/>

**Review by spmonahan — APPROVED** [review]

File: packages/react-components/react-switch/src/components/Switch/useSwitchStyles.ts [hunk 1]
```diff
@@ -50,7 +50,7 @@ const useIndicatorStyles = makeStyles({
     ...shorthands.margin(tokens.spacingVerticalS, tokens.spacingHorizontalS),
     pointerEvents: 'none',
     transitionDuration: '200ms',
```
Comment (spmonahan) [inline packages/react-components/react-switch/src/components/Switch/useSwitchStyles.ts:52]:
```suggestion
    transitionDuration: tokens.durationNormal,
```

File: packages/react-components/react-switch/src/components/Switch/useSwitchStyles.ts [hunk 2]
```diff
@@ -60,7 +60,7 @@ const useIndicatorStyles = makeStyles({
 
     '> *': {
       transitionDuration: '200ms',
```
Comment (spmonahan) [inline packages/react-components/react-switch/src/components/Switch/useSwitchStyles.ts:62]:
```suggestion
    transitionDuration: tokens.durationNormal,
```

File: packages/react-components/react-button/src/components/Button/useButtonStyles.ts [hunk 3]
```diff
@@ -72,7 +72,7 @@ const useRootStyles = makeStyles({
   transition: {
     transitionDuration: '100ms',
```
Comment (spmonahan) [inline packages/react-components/react-button/src/components/Button/useButtonStyles.ts:73]:
```suggestion
    transitionDuration: tokens.durationFaster,
```

**Review by spmonahan — APPROVED** [review]

**Review by sopranopillow — APPROVED** [review]

**Pull request merged**